### ISSUE TITLE
Image::setTargetPath problem with /web/ folder

### ISSUE THREAD

**Issue by aschempp** [contao/core-bundle#337]
Yet another problematic folder. When setting a target folder for an image, the path is relative to `TL_ROOT`. Which might not be available from the web directory.

**Comment by leofeyer** [comment]
You will have to provide the correct path. There is no way to guess as far as I see it.

**Comment by aschempp** [comment]
Yeah, but what's the correct path? It only works with the current path in `TL_ROOT/assets/images` because `assets/images` is also available in the web folder. It's impossible to specify a folder directly in web. We would need to strip the web folder from the output path.

**Comment by leofeyer** [comment]
How do I set up a test case?

**Comment by aschempp** [comment]
Create a folder in `web` (e.g. `/web/images`) and use it as a target folder with `Image::get`. The returned path of `Image::get` will (obviously) be `web/images/…` which will not render in the frontend.

**Comment by leofeyer** [comment]
Fixed in 784ab94587e2178ffa99de0555b004abe22aa13c.

**Comment by Toflar** [comment]
I think this bugfix is wrong.
It works if you use 

```php
\Image::get($arrItem['singleSRC']);
```

but it does obviously not work if you use the class in it's new, object oriented way, like for example `\Controller::addImageToTemplate()` does:

```php
$src = \Image::create($arrItem['singleSRC'], $size)->executeResize()->getResizedPath();
```

So, imho, stripping the `/web` prefix should go into `getResizedPath()`, shouldn't it @contao/developers?

**Comment by aschempp** [comment]
sounds reasonable :+1:

**Comment by ausi** [comment]
I’m not sure if we should handle this at all.

Shouldn’t someone using `Image` that way be aware of stripping the web folder himself?

Is `/web` hardcoded?

**Comment by Toflar** [comment]
Yes, I asked myself this question as well because of the fact that `web` is not a requirement for symfony.

**Comment by bytehead** [comment]
It's not. http://symfony.com/doc/current/cookbook/configuration/override_dir_structure.html#override-the-web-directory

**Comment by bytehead** [comment]
We actually work mostly with symlinks to `/web`. I never had to change `/web` so far.

**Comment by leofeyer** [comment]
It is not a requirement for Symfony but (currently) is for Contao. We could make it dynamic of course, but I don't think it is worth the overhead.

If we decided to handle the case, what about @Toflar's suggestion?

> So, imho, stripping the `/web` prefix should go into `getResizedPath()

**Comment by ausi** [comment]
IMO we should not handle that case and discuss how this should work with the new image services from #342.

In the current draft there is a `getUrl` method which accepts the root directory as the first parameter: https://github.com/madeyourday/contao-core-bundle/commit/22bdf57a2f69d9c33c82a6fa2c02b216325e6af4

**Comment by leofeyer** [comment]
We need to handle the case, otherwise the back end icons would no longer be displayed.

**Comment by ausi** [comment]
Then I must have misunderstood the issue.

If I remove the web-path related code from `Image`, I still see all icons in the backend.

**Comment by leofeyer** [comment]
As discussed in Haag, we want to install the assets directly in the `web/` folder.

**Comment by leofeyer** [comment]
@Toflar Let's assume an 400x400 pixel test image. Is the following correct?

```php
$src = Image::create('web/images/test.jpg', [300, 300, 'crop'])->executeResize()->getResizedPath();
echo $src; // assets/images/3/test-e6a62ef3.jpg -> OK

// No resizing required
$src = Image::create('web/images/test.jpg', [400, 400, 'crop'])->executeResize()->getResizedPath();
echo $src; // web/images/test.jpg -> ERROR

// Legacy method
$src = Image::get('web/images/test.jpg', 400, 400, 'crop');
echo $src; // images/test.jpg -> OK
```

So to restore a consistent behavior, the `Image` class would have to add and strip the `web/` prefix just like the legacy method does, right?

**Comment by leofeyer** [comment]
Please check fc26fe7842c4f623ed961919711636c873b436b1. It should fix the problem.

**Comment by leofeyer** [comment]
Fixed in 31261a5aa33005ac404538d69f7028513144375e.
